classify-split-extract-workflow/classify-job/config.py

```diff
@@ -0,0 +1,350 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import json
+import os
+from typing import Optional, Dict
+
+import google.auth
+from google.cloud import storage
+from logging_handler import Logger
+
+logger = Logger.get_logger(__file__)
+
+# Environment variables and default settings
+PROJECT_ID = os.environ.get("PROJECT_ID")
+if not PROJECT_ID:
+    _, PROJECT_ID = google.auth.default()
+
+CLASSIFY_INPUT_BUCKET = os.environ.get("CLASSIFY_INPUT_BUCKET", f"{PROJECT_ID}-documents")
+CLASSIFY_OUTPUT_BUCKET = os.environ.get("CLASSIFY_OUTPUT_BUCKET", f"{PROJECT_ID}-workflow")
+INPUT_FILE = os.environ.get("INPUT_FILE")
+GOOGLE_APPLICATION_CREDENTIALS = os.environ.get("GOOGLE_APPLICATION_CREDENTIALS")
+OUTPUT_FILE_JSON = os.environ.get("OUTPUT_FILE_JSON", "classify_output.json")
+OUTPUT_FILE_CSV = os.environ.get("OUTPUT_FILE_CSV", "classify_output.csv")
+CALL_BACK_URL = os.environ.get("CALL_BACK_URL")
+BQ_DATASET_ID_PROCESSED_DOCS = os.environ.get("BQ_DATASET_ID_PROCESSED_DOCS", "processed_documents")
+BQ_OBJECT_TABLE_RETENTION_DAYS = int(os.environ.get("BQ_OBJECT_TABLE_RETENTION_DAYS", 7))
+BQ_DATASET_ID_MLOPS = os.environ.get("BQ_DATASET_ID_MLOPS", "mlops")
+BQ_PROJECT_ID = os.environ.get("BQ_PROJECT_ID", PROJECT_ID)
+BQ_REGION = os.environ.get("BQ_REGION", "us")
+BQ_GCS_CONNECTION_NAME = os.environ.get("BQ_GCS_CONNECTION_NAME", "bq-connection-gcs")
+START_PIPELINE_FILENAME = "START_PIPELINE"
+CLASSIFIER = "classifier"
+DOCAI_OUTPUT_BUCKET = os.environ.get("DOCAI_OUTPUT_BUCKET", f"{PROJECT_ID}-docai-output")
+CONFIG_BUCKET = os.environ.get("CONFIG_BUCKET", f"{PROJECT_ID}-config")
+CONFIG_FILE_NAME = "config_4medica.json"
+CLASSIFICATION_UNDETECTABLE = "unclassified"
+CLOUD_RUN_EXECUTION = os.environ.get("CLOUD_RUN_EXECUTION")
+REGION = os.environ.get("REGION")
+SPLITTER_OUTPUT_DIR = os.environ.get("SPLITTER_OUTPUT_DIR", "splitter_output")
+
+PDF_EXTENSION = ".pdf"
+PDF_MIME_TYPE = "application/pdf"
+MIME_TYPES = [
+    PDF_MIME_TYPE,
+    # "image/gif",  # TODO: Add/Check support for all these types
+    # "image/tiff",
+    # "image/jpeg",
+    # "image/png",
+    # "image/bmp",
+    # "image/webp"
+]
+
+NO_CLASSIFIER_LABEL = "No Classifier"
+METADATA_CONFIDENCE = "confidence"
+METADATA_DOCUMENT_TYPE = "type"
+
+FULL_JOB_NAME = f"projects/{PROJECT_ID}/locations/{REGION}/jobs/classify-job"
+
+# Global variables
+gcs = None
+bucket = None
+last_modified_time_of_object = datetime.datetime.now()
+config_data = None
```

Comment:
It's probably not best to use global variables here if you can avoid it.

Reply:
That was done to avoid reloading config.json file stored at gcs. Only if the file was modified.
Borrowing it from another project since it has proven to be working very stable.
Do not know a better way to trade in